src/ledger/test/LedgerTxnTests.cpp

```diff
@@ -2821,245 +2821,176 @@ TEST_CASE("Erase performance benchmark", "[!hide][erasebench]")
-                    successfulFailures++;
-                }
-            }
-            REQUIRE(successfulFailures <= expectedFailedKeys.size());
-        };
+    Config cfg = getTestConfig();
+    cfg.ENTRY_CACHE_SIZE = 10;
 
-    auto runTest = [&](size_t cacheSize) {
-        Config cfg = getTestConfig();
-        cfg.ENTRY_CACHE_SIZE = cacheSize;
+    // Test setup.
+    VirtualClock clock;
+    cfg.EXPERIMENTAL_BUCKETLIST_DB = true;
+    Application::pointer app = createTestApplication(clock, cfg);
+    UnorderedSet<LedgerKey> keysToPrefetch;
+    auto& root = app->getLedgerTxnRoot();
+    LedgerTxn ltx(root);
 
-        // Test setup.
-        VirtualClock clock;
-        cfg.EXPERIMENTAL_BUCKETLIST_DB = true;
-        Application::pointer app = createTestApplication(clock, cfg);
-        UnorderedSet<LedgerKey> keysToPrefetch;
-        auto& root = app->getLedgerTxnRoot();
-        LedgerTxn ltx(root);
+    std::set<LedgerKey> deadKeySet;
 
-        UnorderedSet<LedgerKey> expectedSuccessKeys;
-        UnorderedSet<LedgerKey> expectedFailedKeys;
-        std::set<LedgerEntry> entrySet;
-        std::set<LedgerKey> deadKeySet;
+    auto lkMeterCold = std::make_shared<LedgerKeyMeter>();
+    auto lkMeterHot = std::make_shared<LedgerKeyMeter>();
 
-        auto lkMeterCold = std::make_shared<LedgerKeyMeter>();
-        auto lkMeterHot = std::make_shared<LedgerKeyMeter>();
+    auto contractDataEntry =
+        LedgerTestUtils::generateValidLedgerEntryOfType(CONTRACT_DATA);
+    contractDataEntry.lastModifiedLedgerSeq = 1;
+    ltx.createWithoutLoading(contractDataEntry);
 
-        auto entries =
-            LedgerTestUtils::generateValidUniqueLedgerEntriesWithExclusions(
-                {TTL}, 3);
+    auto classicEntry = LedgerTestUtils::generateValidLedgerEntryOfType(OFFER);
+    classicEntry.lastModifiedLedgerSeq = 1;
+    ltx.createWithoutLoading(classicEntry);
 
-        auto deadKeys =
-            LedgerTestUtils::generateValidUniqueLedgerEntryKeysWithExclusions(
-                {CONFIG_SETTING}, 1);
-        auto ttlEntry = LedgerTestUtils::generateValidLedgerEntryOfType(TTL);
-        entries.push_back(ttlEntry);
-        for (auto e : entries)
-        {
-            e.lastModifiedLedgerSeq = 1;
-            ltx.createWithoutLoading(e);
-            entrySet.emplace(e);
-        };
-        for (const auto& k : deadKeys)
-        {
-            ltx.eraseWithoutLoading(k);
-            deadKeySet.emplace(k);
-        }
+    LedgerEntry TTLEntry;
+    TTLEntry.data.type(TTL);
+    TTLEntry.data.ttl().keyHash = getTTLKey(contractDataEntry).ttl().keyHash;
+    TTLEntry.data.ttl().liveUntilLedgerSeq =
+        contractDataEntry.lastModifiedLedgerSeq + 1;
 
-        // Insert all entries into the database.
-        std::vector<LedgerEntry> ledgerVect{entrySet.begin(), entrySet.end()};
-        std::vector<LedgerKey> deadKeyVect{deadKeySet.begin(),
-                                           deadKeySet.end()};
+    auto deadKeys =
```

Comment:
Instead of a set of size 1, just make the dead key a single CONTRACT_DATA key. This will guarantee uniqueness against the other generated keys and will make our footprint valid.